programs/stake/src/stake_state.rs

```diff
@@ -821,6 +821,10 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
         config: &Config,
         signers: &HashSet<Pubkey>,
     ) -> Result<(), InstructionError> {
+        if vote_account.owner()? != solana_vote_program::id() {
+            return Err(InstructionError::IncorrectProgramId);
+        }
```

Comment:
@jackcmay just making sure; it seems that `IncorrectProgramId` isn't used much; but I think this is a perfect usecase of `IncorrectProgramId` for these kinds of checks, I guess.

Reply:
The description of this error is doesn't make much sense but the way you propose to use it here makes as much sense as any ;-)

Reply:
Fwiw, this seems perfectly consistent with how the error is used in the Vest program

Reply:
yeah, I copied from that. :)